### DISCUSSION TITLE
Some components are missing when js is bundled

### DISCUSSION THREAD

**Discussion by weboose** [breezefront/community#198] [1. Help]
simple example:
```shell
# enable javascript bundle
php bin/magento config:set dev/js/merge_files 1
php bin/magento config:set dev/js/enable_js_bundling 1
php bin/magento config:set dev/js/minify_files 1
php bin/magento config:set dev/css/merge_css_files 1
php bin/magento config:set dev/css/minify_files 1

# delete breeze theme bundles/*
rm -rf ./pub/static/frontend/VENDOR/THEME_NAME/en_US/Swissup_Breeze/bundles/*

# clear cache
php bin/magento cache:flush
```
Open a browser and visit your site, When the page is loaded, many js files will appear in the `Swissup_Breeze/bundles` folder, and these files do not contain all components.

If I want to update my js, I have to use `setup:static-content:deploy` and pause the website access service, Is there a way that I can update my js components and have the bundled js generated correctly?

**Comment by vovayatsyuk** [comment]
Is your site in dev or production mode?

**Reply by weboose** [reply]
production mode.

**Reply by vovayatsyuk** [reply]
Then you must deploy static content using `setup:static-content:deploy` mode. Magento doesn't deploy the scripts on the fly when the store is in production mode. It's weird that bundles are generated for you in this case. I will look into this.

**Reply by weboose** [reply]
To be precise, I want the `setup:static-content:deploy` command to force overwrite the existing js bundle. Currently, if the bundled js already exists, it will not be updated.

**Reply by vovayatsyuk** [reply]
Thanks for the clarification!

**Comment by weboose** [comment]
I have a simple fix for it, modify `Swissup\Breeze\Block\Js.php`
```php
protected function _toHtml()
    {
        $assets = [];

        if ($this->assetConfig->isMergeJsFiles() || $this->assetConfig->isBundlingJsFiles()) {
-            $assets = $this->deployBundledAssets();
+            $assets = $this->getBundledAssets();
        } else {
```

```php
+public function getBundledAssets($jsBuildParams = []): array
+    {
+        $builds = [];
+        foreach ($this->getAllBundles() as $name => $bundle) {
+           $builds[$name] = $this->jsBuildFactory->create(array_merge([
+                    'name' => 'Swissup_Breeze/bundles/' . $name,
+                    'items' => $bundle['items'],
+                ], $jsBuildParams));
+        }
+
+        $assets = [];
+        foreach ($this->getActiveBundles() as $name => $bundle) {
+            $assets = array_merge($assets, $builds[$name]->getBundledAssets());
+        }
+
+        return $assets;
+    }
```

**Comment by vovayatsyuk** [comment]
This [commit](https://github.com/breezefront/module-breeze/commit/b47779a6f46b4c6b031694bc90677dc7480b8007) changes deployment behavior. Now, the bundles will be recreated when running `setup:static-content:deploy`.

**Reply by weboose** [reply]
nice, good job : )
